db/CSV_Writers/Review_Utils/reviewFunctions.js

```diff
@@ -0,0 +1,46 @@
+const { Promise, path, fs } = require('../packages.js');
+const { twoValProb, threeValProb, generateInBetweenSync } = require('../utils/randomMathGeneration');
+const { primaryRecordNumber, reviewsBigBatchLimiter, _usersRecordsNumber } = require('../config/config.js');
+
+const reviewTableKeys = ["game", "userID", "reviewText", "creationDate", "recommended", "helpfulCount", "notHelpfulCount", "helpfulScore", "funnyCount", "earlyAccess", "awards", "comments"].join(",");
+
+mE = module.exports = {
+  paragraphGetter: (numOfPs, paragraphs) => {
+    return [...new Array(numOfPs)].map(num => paragraphs[generateInBetweenSync(1000)]).join('');
+  },
+
+  reviewBatchGenerator: async () => {
+    const paragraphs = require('../Read_Write_Files/reviews.json');
+    const bigBatchStr = [...new Array(reviewsBigBatchLimiter)].reduce((str) => {
+      const helpfulCount = generateInBetweenSync(100);
+      const notHelpfulCount = generateInBetweenSync(10);
+      const creationDate = (1616535988000 - (generateInBetweenSync(31536000000)));
+      const helpfulScore = helpfulCount / (helpfulCount + notHelpfulCount) * 100;
+      return str += `a,${generateInBetweenSync(primaryRecordNumber)},${generateInBetweenSync(_usersRecordsNumber)},${mE.paragraphGetter(generateInBetweenSync(1), paragraphs)},${creationDate},${twoValProb(.15)},${helpfulCount},${notHelpfulCount},${helpfulScore},${threeValProb(.7, .8)},${threeValProb(.7, .9)},${twoValProb(.9)},${generateInBetweenSync(20)}\r\n`;
+    }, '');
+    fs.writeFileSync("Read_Write_Files/oneMillionReviews.csv", `${reviewTableKeys}\r\n${bigBatchStr}`);
+  },
+
+  reviewDBQuery: () => {
+    const db = Promise.promisifyAll(require('../../../db'));
+    return new Promise((resolve, reject) => {
+      db.query(
+        `LOAD DATA LOCAL INFILE 'Read_Write_Files/oneMillionReviews.csv'
+        INTO TABLE reviews
+        FIELDS TERMINATED BY ','
+        ENCLOSED BY '"'
+        LINES TERMINATED BY '\n'
+        IGNORE 1 ROWS;`
+        , (err, result) => {
+          if (err) {
+            console.log(err);
+            return reject("db", `${err.message}`);
+          }
+          resolve();
+        })
+    })
+  },
```

Comment:
very cool function. a couple followup questions/comments:
1. what's going on with the promisifyAll()? it looks like youre promisifying an entire folder. ive have not seen this before.
2. is there a way to make your code more DRY? it looks like reviewDBQuery and userDBQuery are almost identical. i think you would be able to add some parameters to the function and make it reusable

Reply:
Hey Max thank for the great feedback! Ok so in terms of promisfyAll() this was some logic that I inherited from Tim and i'm not yet 100% sure about how it works, and i'm unclear if I take it out if it breaks the script or not. In terms of the require statement, somehow when I require the whole database folder it imports the index.js file in the root of the db folder only- it would be the same as requiring db/index.js  for example. That index.js file holds the mysql configuration and connection function.

Also 100% on making the DBQuery function resuable, that is the first thing i'm going to do after responding here. I'm curious what you mean by dry though? To me it seems that doing so would make things more modular and dynamic (which definitely scratches my itch).